Automattic/wp-calypso
# Automattic for Agencies: Implement license details actions

Resolves https://github.com/Automattic/jetpack-genesis/issues/263

## Proposed Changes

<img width="1617" alt="Screenshot 2024-02-29 at 10 36 45 AM" src="https://github.com/Automattic/wp-calypso/assets/10586875/f20fb3fd-0454-4640-ab1c-ed1ac4614a58">

This PR implements:

- Actions for the details view
- Mock API for revoke license and download URL

NOTE: 

- The license data is mocked in the UI currently. It will be loaded only in the dev environment. It will not load until the feature flag is enabled on other ENVs. 
- Debug URL will be implemented later. 
- Ignore the button styles, it will be fixed via a different PR later.

## Testing Instructions

- Switch to the branch `git checkout add/implement-a4a-license-details-actions`.
- Start the server by running `yarn start-a8c-for-agencies`.
- Click the Purchases menu item > Verify that you can see the Licenses page with 2 licenses. Expand the first one and verify that the detailed view for the bundle is loaded with a list of child licenses > Expand any child licenses
- Click "Download" and verify you are directed `/test`. Please note this will be fixed when we implement the actual API.
- Click "View Site" and verify that you are redirected to the site URL.
- Click the `Revoke` button > Verify that the confirmation popup is displayed as shown below. Please note clicking on the Review License button doesn't revoke the license, this will be implemented on another PR.

<img width="702" alt="Screenshot 2024-02-29 at 10 39 56 AM" src="https://github.com/Automattic/wp-calypso/assets/10586875/5ac0f77d-f5d2-4222-95a0-e33a03f61168">

## Pre-merge Checklist
